internal/engine/ast.go

```diff
@@ -449,57 +447,33 @@ func buildExprImpl(cur *exprpb.Expr, acc *enginev1.PlanResourcesFilter_Expressio
 		}
 		acc.Node = mkExprOpExpr(Struct, operands...)
 	case *exprpb.Expr_ComprehensionExpr:
-		x := expr.ComprehensionExpr
-		var operator string
-		var step *exprpb.Expr_CallExpr
-		var ok bool
-		if step, ok = x.LoopStep.ExprKind.(*exprpb.Expr_CallExpr); !ok {
-			return fmt.Errorf("expected loop-step expression type CallExpr, got: %T", x.LoopStep.ExprKind)
-		}
-		var le *exprpb.Expr
-		switch step.CallExpr.Function {
-		case operators.LogicalAnd:
-			operator = All
-			le = step.CallExpr.Args[1]
-		case operators.LogicalOr:
-			operator = Exists
-			le = step.CallExpr.Args[1]
-		case operators.Add:
-			operator = Map
-			if elements := step.CallExpr.Args[1].GetListExpr().GetElements(); len(elements) > 0 {
-				le = elements[0]
-			}
-		case operators.Conditional:
-			switch x.AccuInit.ExprKind.(type) {
-			case *exprpb.Expr_ListExpr:
-				operator = Filter
-			case *exprpb.Expr_ConstExpr:
-				operator = ExistsOne
-			default:
-				return fmt.Errorf("expected loop-accu-init expression type ConstExpr or ListExpr, got: %T", x.AccuInit.ExprKind)
+		lambdaAst, err := BuildLambdaAST(expr.ComprehensionExpr)
+		iterRange := lambdaAst.IterRange
```

Comment:
I think this would panic if `BuildLambdaAST` returns `nil, err` - we should probably move the error handling up above this line.